tests/test_grid.py

```diff
@@ -135,6 +141,28 @@ def test(sym, endpoint, axis, ntheta_is_odd):
         test(sym=True, endpoint=True, axis=False, ntheta_is_odd=False)
         test(sym=True, endpoint=True, axis=True, ntheta_is_odd=False)
 
+    @pytest.mark.unit
+    def test_linear_grid_symmetric_nodes_consistency(self):
+        """Test that specifying theta nodes from [0, pi] is sufficient."""
+        # uniform spacing
+        theta = np.linspace(0, 2 * np.pi, 10)
+        lg_1 = LinearGrid(L=5, theta=theta, N=5, sym=True)
+        lg_2 = LinearGrid(L=5, theta=theta[theta <= np.pi], N=5, sym=True)
+        np.testing.assert_allclose(lg_1.nodes, lg_2.nodes)
+        np.testing.assert_allclose(lg_1.spacing, lg_2.spacing)
+        np.testing.assert_allclose(lg_1.weights, lg_2.weights)
+
+        # non-uniform spacing
```

Comment:
what exactly is the expected behavior for a linear grid with non-uniformly spaced points?

Reply:
The rule used to compute the spacing when an array is given is: $d\theta$ is chosen to be half the cyclic distance of the surrounding two nodes. In other words, if we parameterize a circle’s perimeter from 0 to $2\pi$ , and place points on it according to the given array (e.g. theta = np.linspace(0, 2pi, 4)), then the $d\theta$ assigned to each node will be half the parameterized distance along the arc between its left and right neighbors. The process is the same for $d\zeta$.

Visuals:
Here we are visualizing the $d\theta$ spacing of a $\theta$ curve (intersection of $\rho$ and $\zeta$ surface).
Let the node's coordinates be at the values given by the filled circles.
The $d\theta$ spacing assigned to each node is the length of arc of the same color.

Example: uniform spacing
![linearly spaced](https://user-images.githubusercontent.com/98659578/228340341-fdf05f7a-71dc-4e6f-a901-b6346567de83.png)

Example: non-uniform spacing
![non linear spacing](https://user-images.githubusercontent.com/98659578/228341003-d5a51b93-5006-4e11-a640-c2bf7e42c8e0.png)

Example: Two nodes with symmetry set to False
![another example](https://user-images.githubusercontent.com/98659578/228341082-ebcd15b9-a28a-47c0-b57b-287d8a6764c4.png)

Example: The same two nodes with symmetry set to True
Notice now the red node is given more weight, because symmetry says there is a duplicate of that node (in black) across the axis of symmetry.
![symmetry](https://user-images.githubusercontent.com/98659578/228341186-21743469-53ea-4610-af35-54fceb41a8cd.png)

Reply:
In this sense LinearGrid can actually work better than the base Grid for custom grids. User can give nodes at whatever location when they pass in arrays, and the spacing will be a midpoint rule. In Grid it's just assumed constant.

Reply:
Ok. I don't see the use case of calling `LinearGrid` with a non-uniform spacing in theta/zeta but this seems a good way to treat that case. Thanks for the explanation